pkg/apis/pipeline/v1alpha1/pipelineparams_validation.go

```diff
@@ -0,0 +1,107 @@
+/*
+Copyright 2018 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha1
+
+import (
+	"fmt"
+	"net/url"
+
+	"github.com/knative/pkg/apis"
+	"k8s.io/apimachinery/pkg/api/equality"
+)
+
+func (p *PipelineParams) Validate() *apis.FieldError {
```

Comment:
- For webhook to validate `PipelineParams ` objects, you need to update its handlers like [this](https://github.com/knative/build-pipeline/blob/master/cmd/webhook/main.go#L87). 

- Also under `PipelineParamsSpec ` you need to define `Generation` variable for webhook server not to crash. eg : https://github.com/knative/build-pipeline/blob/master/pkg/apis/pipeline/v1alpha1/task_types.go#L35
This will take care of bumping generations for that respective object.

- PipelineParams need to comply with [these](https://github.com/knative/build-pipeline/blob/master/pkg/apis/pipeline/v1alpha1/task_types.go#L45-L49) interfaces.

Reply:
Thanks @shashwathi , I added those in :)